### DISCUSSION TITLE
Healthcheck and Metrics API

### DISCUSSION THREAD

**Discussion by MichalisDBA** [9tigerio/db2rest#653] [Ideas]
Add a healthcheck API endpoint to get the status of the db2rest instance. Also metrics would be great.

For the healthcheck a simple `status: ok` would be sufficient.

![image](https://github.com/kdhrubo/db2rest/assets/68512501/a85c4e24-ac7e-4c63-b91a-8aaa1df73a5b)

For the metrics we can have some more verbose information about java heap allocation memory, response times, quieries statistics (count, average response times), last 10 errors or warnings etc.

Similar

https://docs.n8n.io/hosting/logging-monitoring/monitoring/#enable-metrics-and-healthz-for-self-hosted-n8n

**Comment by kdhrubo** [comment]
@MichalisDBA - this is already implemented hence closing
[https://db2rest.com/docs/category/actuator](https://db2rest.com/docs/category/actuator)
